### PR TITLE
Add "fork" method on DenoPermissions, PermissionState

### PR DESCRIPTION
**Pull request by humancalico**
For #4779

### CODE REVIEW

File: cli/permissions.rs [hunk 1]
```diff
@@ -57,6 +57,21 @@ impl PermissionState {
     }
     *self
   }
+
+  pub fn fork(
+    self,
+    permission_to_be_forked: bool,
```
Comment (bartlomieju) [inline cli/permissions.rs:63]: Nitpick: rename `permission_to_be_forked` to `value`

File: cli/permissions.rs [hunk 2]
```diff
@@ -735,4 +802,42 @@ mod tests {
...
     drop(guard);
   }
+
+  #[test]
+  fn test_fork() {
+    let guard = PERMISSION_PROMPT_GUARD.lock().unwrap();
+    let perms0 = Permissions::from_flags(&Flags {
+      ..Default::default()
+    });
+    set_prompt_result(true);
```
Comment (bartlomieju) [inline cli/permissions.rs:812]: Add another test case in the same function that uses `set_prompt_result(false);`

**Comment by bartlomieju** [comment]
@humancalico can you resolve the conflicts? I'll try to land this PR for 1.0.3

**Comment by humancalico** [comment]
Done @bartlomieju

**Pull request merged**